serpentTools/tests/test_microxs.py

```diff
@@ -0,0 +1,182 @@
+        6. test_getXS:
+                        Check the getXS method implemented
+                        to obtain cross-sections for a specific
+                        isotope and reaction. Returns empty list
+                        for non existing isotope and reaction
+            Raises SerpentToolsException:
+                        non-existing universe or wrong
+                        isotope formatting
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.file = os.path.join(TEST_ROOT, 'ref_mdx0.m')
+        with rc:
+            rc['microxs.getFY'] = True
+            rc['microxs.getXS'] = True
+            rc['microxs.getFlx'] = True
+            cls.reader = MicroXSReader(cls.file)
+            cls.reader.read()
+
+    def test_emptyAttributes(self):
+        """Verify that results are not all empty"""
+        mdxFile = os.path.join(TEST_ROOT, 'ref_mdx0_noXS.m')
+        with rc:
+            rc['microxs.getFY'] = False  # do not store fission yields
+            mdxReader = MicroXSReader(mdxFile)
+            with self.assertRaises(SerpentToolsException):
+                mdxReader.read()
+
+    def test_univNum(self):
+        """Test the expected number of universes."""
+        expectedUniv = {'0'}
+        expectedKeys = set(expectedUniv)
+        actualKeys = set(self.reader.xsVal.keys())
+        self.assertSetEqual(expectedKeys, actualKeys)
+
+    def test_fluxdata(self):
+        """Test that flux ratios and uncertainties are properly stored."""
+        univ = '0'
+        expectedValue = [1.60327E+00, 1.59830E+00, 1.59005E+00, 1.57731E+00,
+                         1.55586E+00, 1.54909E+00, 1.54820E+00, 1.53829E+00,
+                         1.55250E+00, 1.54046E+00, 1.54549E+00, 1.55092E+00,
+                         1.54549E+00, 1.53785E+00, 1.54761E+00, 1.54864E+00,
+                         1.53520E+00, 1.52489E+00, 1.49595E+00, 1.51506E+00,
+                         1.50824E+00, 1.50808E+00, 1.48335E+00, 1.45860E+00]
+
+        expectedUnc = [0.00351, 0.00086, 0.00052, 0.00025, 0.00023, 0.00020,
+                       0.00013, 0.00015, 0.00011, 0.00013, 0.00011, 0.00012,
+                       0.00015, 0.00016, 0.00012, 0.00019, 0.00018, 0.00026,
+                       0.00023, 0.00032, 0.00045, 0.00094, 0.00115, 0.00285]
+
+        numpy.testing.assert_equal(self.reader.fluxRatio[univ], expectedValue)
+        numpy.testing.assert_equal(self.reader.fluxUnc[univ], expectedUnc)
+
+    def test_fydata(self):
+        """Test fission yields for the reader."""
+        parent = 922350
+        energy = 2.53000E-08
+        fp = 531350
+        excpectedIndYield = numpy.array([2.92737E-02])
+        excpectedCumYield = numpy.array([6.28187E-02])
+        actualIndYield, actualCumYield = self.reader.getFY(parent, energy, fp)
+        numpy.testing.assert_equal(excpectedIndYield, actualIndYield)
+        numpy.testing.assert_equal(excpectedCumYield, actualCumYield)
+
+    def test_getFY(self):
```

Comment:
If you are doing a single test method that has multiple test cases, e.g. multiple ways the test could fail, then the test method should have some error messages. These can be passed to the `self.assertX` methods with the `msg=<failure message`, and the `assert_<array testing>` with `err_msg=<failure message>`. This will help us isolate the specific failure point.